### PR TITLE
Makefile, basic CI setup, updated README

### PR DESCRIPTION
**Pull request by mcharytoniuk**
I figured out build scripts and created Makefile to streamline the build a bit. Makefile is used inside `.drone.yml` to perform builds, there are also a few new `env` variables to parametrize build. README is updated to reflect changes.

Please let me know if all looks fine. If it does, we then could generate 3.5 release and I would be able to continue working on CI. If CI build fails within this PR, please ignore it, as CI setup is not finished yet. Those changes are here to help me to finish that setup.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# EditorConfig helps developers define and maintain consistent coding styles between different editors and IDEs.
```
Comment (prologic) [inline .editorconfig]: I probably wouldn't commit this file to be honest as it makes an assumption about editor choices that are usually highly opinionated and personal preference :) But up to you!

**Comment by prologic** [comment]
LGTM Will let others review/approve

File: .editorconfig (hunk 1 above)
Comment (mcharytoniuk) [reply .editorconfig]: @prologic Thanks. I'll wait for others, maybe we will figure something out together to stay consistent in the future once and for all. :D

Comment (prologic) [reply .editorconfig]: Hahaha As I like to say:

> You can rip Vim out of my cold dead hands

:)

**Comment by prologic** [comment]
Love that you're using Drone here too! :)

File: .editorconfig (hunk 1 above)
Comment (wawrzek) [reply .editorconfig]: three letters: vim ;-)

Comment (wawrzek) [reply .editorconfig]: I find this file useful as a guideline for any editor.

**Comment by mcharytoniuk** [comment]
@lag-linaro squashing PR before merge would be fine for you?

**Comment by mcharytoniuk** [comment]
@lag-linaro also I think it would be nice to create `CONTRIBUTING` (https://help.github.com/en/articles/setting-guidelines-for-repository-contributors) document and specify all commit / branching rules to setup some guidelines and expectations upfront. After I did those changes it would be a bit problematic to override git history and others who might want to contribute might run into the same problem. Would you like to prepare such document?

**Comment by lag-linaro** [comment]
> @lag-linaro also I think it would be nice to create `CONTRIBUTING` (https://help.github.com/en/articles/setting-guidelines-for-repository-contributors) document and specify all commit / branching rules to setup some guidelines and expectations upfront. After I did those changes it would be a bit problematic to override git history and others who might want to contribute might run into the same problem. Would you like to prepare such document? I think you would be the right person to do so.

I too think a `CONTRIBUTING` document would be a good idea, and would be happy to help review such a document.

Why is it problematic to re-write Git history for unmerged patches?

**Comment by lag-linaro** [comment]
This would make an excellent base for a `CONTRIBUTING` document:

  https://www.kernel.org/doc/html/v4.10/process/development-process.html

5.3 is certainly worth of note:

  https://www.kernel.org/doc/html/v4.10/process/5.Posting.html

**Comment by mcharytoniuk** [comment]
@lag-linaro ok, I'll close this PR and split it into several smaller ones. I'll start separate one with `.editorconfig` and `CONTRIBUTING` to make discussion easier.

**Comment by lag-linaro** [comment]
You don't need to close or split the PR, just update the patches.

Patches should:
-   Be Small
-   Be grouped by functionality
-   Solve one problem OR one enhancement
-   Contain suitable (descriptive, useful) subject lines and commit logs

**Comment by mcharytoniuk** [comment]
@lag-linaro it's fine. It would be easier for me to spli it, also considering that few extra topics emerged. I'll update patches by the way of doing that.

**Comment by mcharytoniuk** [comment]
@lag-linaro ok, done. Please take a look if it looks reasonable now. We can work on merging changes in smaller chunks now. I also changed commit messages.

https://github.com/cruxlinux/docker-crux/pull/20
https://github.com/cruxlinux/docker-crux/pull/19
https://github.com/cruxlinux/docker-crux/pull/18
https://github.com/cruxlinux/docker-crux/pull/17
https://github.com/cruxlinux/docker-crux/pull/16

**Pull request closed without merging**